### ISSUE TITLE
As a cif user, I want to cancel changes to a project, so that I can undo my work

### ISSUE THREAD

**Issue by matthieu-foucault** [bcgov/cas-cif#96]
#### Description:
A user should be able to revert any changes they have made while a project revision is in draft.
Reverting committed changes should not be possible. Since those changes were approved by an admin and committed, a new revision should be done at that point.
The user should be able to revert all changes related to the current revision.

#### Acceptance Criteria:
Given that I have made changes to a project revision
When I want to undo my changes
Then I can revert the changes related to the current revision.

#### Development Checklist:
- [x] The form_change and project_revision tables have `delete` mutations to populate the `deleted_at` / `deleted_by` columns in the database
- [x] The project_revision `delete` mutation should cascade to set the `deleted_at` / `deleted_by` columns of all related `form_change` records
- [x] A `Discard Changes` button is shown on the edit project page as per the wireframe
- [x] https://github.com/graphile-contrib/pg-omit-archived is used to automatically omit deleted records
- [ ] Clicking the `Discard Changes` button shows a list of all changes related to the current revision to be discarded
- [x] Confirming the discard reverts the changes using the mutations described above
- [x] Wireframe completed and linked to ticket
- [x] Meets DOD

#### Personas:
- [Program Development Team](https://bcgov.sharepoint.com/:b:/t/00608-CIFInternalUsers/EcVHGc4X8mhIvQHGSaGJmyAB9F1iBe7MSH1Y1fBKUc1skA?e=fzATIc)
- [Operations Team](https://bcgov.sharepoint.com/:b:/t/00608-CIFInternalUsers/EaswKcprmuNKpEuY29a-Os0BG-h5u-Ztou7UuzK3x0xfjw?e=e8oR1i)
- [Tech Team](https://bcgov.sharepoint.com/:b:/t/00608-CIFInternalUsers/Ec2AmUecYCdDlmURctGwTbMBUMGZXlUyyt5HQZJdpjeejA?e=0axXlm)

**Comment by Sriram-pc** [comment]
Adding applicable Personas to the ticket
[Program Development Team - CIF Persona](https://bcgov.sharepoint.com/:b:/t/00608-CIFInternalUsers/EcVHGc4X8mhIvQHGSaGJmyAB9F1iBe7MSH1Y1fBKUc1skA?e=fzATIc)

[Operations Team - Persona](https://bcgov.sharepoint.com/:b:/t/00608-CIFInternalUsers/EaswKcprmuNKpEuY29a-Os0BG-h5u-Ztou7UuzK3x0xfjw?e=e8oR1i)

[Tech Team - CIF Persona](https://bcgov.sharepoint.com/:b:/t/00608-CIFInternalUsers/Ec2AmUecYCdDlmURctGwTbMBUMGZXlUyyt5HQZJdpjeejA?e=0axXlm)

**Comment by rajpersram** [comment]
see buttons at the bottom of the wireframe
![CIF Workflow Co-Design - Create or Edit Project - Overview tab.jpg](https://images.zenhubusercontent.com/5c2e4cfdab4f3464fb8df8d1/d96fe24f-f278-46a0-b7c7-f57b83e81d75)

**Comment by matthieu-foucault** [comment]
@dleard I think that the description as-is is a bit advanced for an MVP. I'd only have a single `Discard Changes` button, as shown in Raj's wireframe. If there's a need we can add some more granular controls later

**Comment by LindsayMacfarlane** [comment]
Thank you @pbastia ! I tested in Dev by:

- Entering some project details then selecting the 'discard changes button'.
- After selecting button, user is brought back to the project lists table.
- My discarded project was not included in list (as expected) and the 'Add a Project' button displayed (vs. resume project draft).

I did notice that when I discarded the changes, I did not get a list of what was being discarded, but my assumption is this will be a new user story to be done during the new design. 

Good work!
